src/Components/ADT3DSceneBuilder/Internal/Behaviors/BehaviorsForm.tsx

```diff
@@ -128,6 +132,16 @@ const SceneBehaviorsForm: React.FC<IADT3DSceneBuilderBehaviorFormProps> = ({
         selectedBehaviorPivotKey,
         setSelectedBehaviorPivotKey
     ] = useState<BehaviorPivot>(BehaviorPivot.elements);
+    const [
```

Comment:
shouldn't this state be internal to the VisualRuleForm instead of passing the `isPropertyTypeDropdownEnabled `and `setisPropertyTypeDropdownEnabled` handler both as props to it? based on the value of the property the form would know that if it is initially intellisense or not right?

Reply:
It is important to have it in the BehaviorsForm since VisualRuleForm renders inside the container of BehaviorsForm. Styles in BehaviorsForm need that information to handle list sizing correctly.

Reply:
What if we only notify the parent of changes and don't hold the official state up there?

Reply:
Then how would the parent re-render?